python/private/py_executable.bzl

```diff
@@ -591,15 +591,78 @@ def _create_venv(ctx, output_prefix, imports, runtime_details):
         },
         computed_substitutions = computed_subs,
     )
+    site_packages_symlinks = _create_site_packages_symlinks(ctx, site_packages)
 
     return struct(
         interpreter = interpreter,
         recreate_venv_at_runtime = not venvs_use_declare_symlink_enabled,
         # Runfiles root relative path or absolute path
         interpreter_actual_path = interpreter_actual_path,
-        files_without_interpreter = [pyvenv_cfg, pth, site_init],
+        files_without_interpreter = [pyvenv_cfg, pth, site_init] + site_packages_symlinks,
     )
 
+def _create_site_packages_symlinks(ctx, site_packages):
+    """Creates symlinks within site-packages.
+
+    Args:
+        ctx: current rule ctx
+        site_packages: runfiles-root-relative path to the site-packages directory
+
+    Returns:
+        {type}`list[File]` list of the File symlink objects created.
+    """
+
+    # maps site-package symlink to the runfiles path it should point to
+    entries = depset(
+        # NOTE: Topological ordering is used so that dependencies closer to the
+        # binary have precedence in creating their symlinks. This allows the
+        # binary a modicum of control over the result.
```

Comment:
I like this, this is really powerful and acts as I would expect something like this to work.

Maybe it would be worth documenting this in the deps or the newly added site packages root attr?

I see that you've already done this, which makes me warry that the description/behaviour may get out of sync. Is it possible to add a note about the exeisting docs elswhere?

Reply:
Good point. I'll update the deps and site_packages_root attribute to mention the ordering behavior.

Reply:
Done. Add doc to the deps attribute and a mention in the site_packages_root attribute.
LMK if there are additional doc sites it should be mentioned at.

Reply:
I'm not completely sure whether it's relevant here, but since it's so important if it does, please be aware of this legendary comment from the Bazel codebase: https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/analysis/SymlinkEntry.java;l=33?q=Symlinkentr&ss=bazel%2Fbazel